es6-lib/decoders/shapefile.js

```diff
@@ -164,18 +171,35 @@ class Shapefile extends Duplex {
     return emitter;
   }
 
+  _hasRequiredComponents(components) {
+    var groups = this._groupComponents(components);
+    return _.reduce(groups, (errors, [shp, _prj, dbf]) => {
+      //if the shp or dbf are undefined, then record the error. prj
+      //is technically optional
+      if(!shp) {
+        errors.push(new Error('Missing spatial (.shp) file.'));
+      }
+      if(!dbf) {
+        errors.push(new Error('Missing attributes (.dbf) file.'));
+      }
+      return errors;
+    }, []);
+  }
+
   _groupComponents(components) {
     components.sort();
     var shps = components.filter((c) => isShp(c));
     var projs = components.filter((c) => isProjection(c));
-    return _.zip(shps, projs);
+    var dbfs = components.filter((c) => isDbf(c));
+    return _.zip(shps, projs, dbfs);
   }
 
   _startPushing() {
     this._isPushing = true;
 
     var groups = this._groupComponents(this._components);
-    async.mapSeries(groups, ([shp, proj], cb) => {
+
+    async.mapSeries(groups, ([shp, proj, _dbf], cb) => {
```

Comment:
it means that the variable is unused, but it's there to make the shape of the datum clear